code/modules/admin/view_variables/view_variables.dm

```diff
@@ -14,26 +14,48 @@
 	var/datum/asset/asset_cache_datum = get_asset_datum(/datum/asset/simple/vv)
 	asset_cache_datum.send(usr)
 
+	var/isappearance = isappearance(thing)
 	var/islist = islist(thing) || (!isdatum(thing) && hascall(thing, "Cut")) // Some special lists dont count as lists, but can be detected by if they have list procs
-	if(!islist && !isdatum(thing))
+	if(!islist && !isdatum(thing) && !isappearance)
 		return
 
 	var/title = ""
 	var/refid = REF(thing)
 	var/icon/sprite
 	var/hash
 
-	var/type = islist? /list : thing.type
+	var/type = islist? /list : (isappearance ? "/appearance" : thing.type)
 	var/no_icon = FALSE
 
 	if(isatom(thing))
 		sprite = getFlatIcon(thing)
 		if(!sprite)
 			no_icon = TRUE
 
-	else if(isimage(thing))
+	else if(isimage(thing) || isappearance)
+		#define ICON_STATE_CHECKED 1
+		#define ICON_STATE_NULL 2
```

Comment:
```suggestion
#define ICON_STATE_CHECKED 1
#define ICON_STATE_NULL 2
```

i know it's ugly but we don't want to imply that these defines have any type of scope or anything like that, because they don't. maybe you should move the defines to the start of the proc after unindenting them?